src/swaps/blindex/BlindexSwapProvider.js

```diff
@@ -0,0 +1,509 @@
+import BN from 'bignumber.js'
+import { v4 as uuidv4 } from 'uuid'
+import JSBI from 'jsbi'
+import * as ethers from 'ethers'
+
+import UniswapV2Router from '@uniswap/v2-periphery/build/IUniswapV2Router02.json'
+import { Token, CurrencyAmount, Fraction, Percent } from '@uniswap/sdk-core'
+import ERC20 from '@uniswap/v2-core/build/ERC20.json'
+import { ChainId, chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import cryptoassets from '@/utils/cryptoassets'
+import { ChainNetworks } from '@/utils/networks'
+
+import { isERC20 } from '../../utils/asset'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils'
+import { SwapProvider } from '../SwapProvider'
+import { BestRouteService } from './BestRouteService'
+import { blindexConfig } from './BlindexConfig'
+
+const SWAP_DEADLINE = 30 * 60 // 30 minutes
+const DECIMALS = 18
+const RSK_RPC_URL = 'https://public-node.rsk.co'
+
+class BlindexSwapProvider extends SwapProvider {
+  constructor(config) {
+    super(config)
+    this._apiCache = {}
+    this.bestRouteService = new BestRouteService()
+    this.wrbtcAddress = blindexConfig.ERC20_INFO.find((x) => x.symbol === 'WRBTC').address
+    this.nativeTokenSymbol = chains[ChainId.Rootstock].nativeAsset //RBTC
+    this.availableTokenSymbols = blindexConfig.ERC20_INFO.filter((entry) => entry !== 'UNI-V2').map(
+      (entry) => (entry.symbol === 'WRBTC' ? this.nativeTokenSymbol : entry.symbol)
+    )
+  }
+
+  async getSupportedPairs() {
+    return []
+  }
```

Comment:
Is this here by mistake? I see it's empty and not being used

Reply:
That's how almost every swap provider is implemented. The Liquality and FastBtc swap provider have them defined but as far as I can see the data is then saved in the application store for later use.
And it is later used to feed data to Liquality provider swap. So I suppose it serves a purpose of caching the data for easier access.
![image](https://user-images.githubusercontent.com/91085805/161045319-8b625e74-d121-436a-8b93-cc99f8c673d2.png)